### PR TITLE
Extract name recommendation logic into a separate type

### PR DESCRIPTION
**Pull request by genlu**
This just refactor the code w/o any change to the logic. I have extracted the code for declaration name recommendation into a separate type. The goal is to make name recommender an extension point so IntelliCode can provide suggestion as well, which will be in a follow up PR.

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(DeclarationNameCompletionProvider), LanguageNames.CSharp)]
+    [ExtensionOrder(After = nameof(TupleNameCompletionProvider))]
+    [Shared]
+    internal partial class DeclarationNameCompletionProvider : LSPCompletionProvider
+    {
+        private ImmutableArray<DeclarationNameRecommender> Recommenders { get; }
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:27]: readonly?

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            return CommonCompletionItem.Create(
+                name,
+                displayTextSuffix: "",
+                CompletionItemRules.Default,
+                glyph: glyph,
+                sortText: sortText,
+                description: CSharpFeaturesResources.Suggested_name.ToSymbolDisplayParts());
+        }
+    }
+}
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:113]: is this all a move? did anything hange?

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs [hunk 3]
```diff
@@ -1,626 +0,0 @@
...
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.CSharp.Extensions;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Editing;
-using Microsoft.CodeAnalysis.LanguageService;
-using Microsoft.CodeAnalysis.Shared.Extensions;
-using static Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SymbolSpecification;
-
-namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs:16]: very confused why github didn't pick upt he rename.  is this substantively chnaged at all?

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                    if (context.SemanticModel.GetSymbolInfo(context.TargetToken).GetAnySymbol() is null)
+                    {
+                        return;
+                    }
+                }
+
+                var nameInfo = await NameDeclarationInfo.GetDeclarationInfoAsync(document, position, cancellationToken).ConfigureAwait(false);
+                using var _ = ArrayBuilder<(string name, Glyph glyph)>.GetInstance(out var result);
+
+                var sortValue = 0;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:76]: unused?

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:76]: move to where it is needed.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs (hunk 1 above)
Comment (genlu) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:27]: Already a readonly property

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs (hunk 2 above)
Comment (genlu) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:113]: Yes. The only change (except simply moving code around) is extracting code into methods.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs:27]: d'oh!

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by ryzngard — APPROVED** [review]
I'm going to trust you didn't change anything

**Pull request merged**